src/corelibs/U2Formats/src/FastqFormat.cpp

```diff
@@ -357,16 +410,29 @@ Document* FastqFormat::loadDocument(IOAdapter* io, const U2DbiRef& dbiRef, const
     CHECK_EXT(io != NULL && io->isOpen(), os.setError(L10N::badArgument("IO adapter")), NULL);
     QVariantMap hints = _hints;
     QList<GObject*> objects;
+    QMap<QString, QString> skippedLines;
 
     int gapSize = qBound(-1, DocumentFormatUtils::getMergeGap(_hints), 1000*1000);
     int predictedSize = qMax(100*1000, DocumentFormatUtils::getMergedSize(hints, gapSize==-1 ? 0 : io->left()));
 
     QString lockReason;
-    load(io, dbiRef, _hints, io->getURL(), objects, os, gapSize, predictedSize, lockReason);
+    load(io, dbiRef, _hints, io->getURL(), objects, os, gapSize, predictedSize, lockReason, skippedLines);
```

Comment:
I would wrap it into a new method (processSkippedLines(), processWarnings() etc.).